### ISSUE TITLE
Question of the implementation

### ISSUE THREAD

**Issue by Christinepan881** [facebookresearch/Mask2Former#77]
Hi, can I ask a question about the transformer decoder implementation?

In the VideoMultiScaleMaskedTransformerDecoder class, why this line?
`attn_mask[torch.where(attn_mask.sum(-1) == attn_mask.shape[-1])] = False` 
It means for the query without attended part, change all positions for the query to attend. But I didn't get the intuition for that.

Thanks!

**Comment by bowenc0221** [comment]
Otherwise the output will be `NaN`.

**Comment by 9p15p** [comment]
This line of code may mean:
If some attn_mask are all `True`, then change them all into  `False`

**Comment by victor000000** [comment]
<img width="350" alt="image" src="https://user-images.githubusercontent.com/1636580/182786711-21b19a7c-dbaa-4ad9-a26c-bc81c9687185.png">
softmax(-inf) -> NaN
